ngraph/frontend/onnx_import/src/editor/editor.cpp

```diff
@@ -369,3 +403,118 @@ void onnx_import::ONNXModelEditor::set_input_values(
+    }
+
+    // 2. get internal tensors via intersection of inputs and outputs.
+    std::vector<std::string> internal_tensors;
+
+    std::set<std::string> inputs_set(inputs.begin(), inputs.end());
+    std::set<std::string> outputs_set(outputs.begin(), outputs.end());
+
+    std::set_intersection(inputs_set.begin(),
+                          inputs_set.end(),
+                          outputs_set.begin(),
+                          outputs_set.end(),
+                          std::inserter(internal_tensors, internal_tensors.end()));
+
+    auto is_external_tensor = [&node_indexes, &nodes](std::string tensor_name) {
+        bool is_external = false;
+        for (int i = 0; i < nodes.size(); ++i)
+        {
+            if (std::count(node_indexes.begin(), node_indexes.end(), i))
+            {
+                continue;
+            }
+
+            is_external = is_external ||
+                          std::count(nodes[i].input().begin(), nodes[i].input().end(), tensor_name);
+        }
+        return is_external;
+    };
+
+    auto is_on_kill_list = [&internal_tensors](std::string target_name) {
+        return std::count(internal_tensors.begin(), internal_tensors.end(), target_name);
+    };
+
+    // 3. remove internal tensors from input list.
+    auto new_end = std::remove_if(inputs.begin(), inputs.end(), is_on_kill_list);
+    inputs.erase(new_end, inputs.end());
+
+    // 4. remove from internal tensors a tensor which is used outside of replaced graph
+    new_end = std::remove_if(internal_tensors.begin(), internal_tensors.end(), is_external_tensor);
+    internal_tensors.erase(new_end, internal_tensors.end());
+
+    // 5. remove internal tensors from output list.
+    new_end = std::remove_if(outputs.begin(), outputs.end(), is_on_kill_list);
+    outputs.erase(new_end, outputs.end());
+
+    // 6. outputs have to be unique
+    new_end = std::unique(outputs.begin(), outputs.end());
+    outputs.erase(new_end, outputs.end());
+
+    // 7. replace the first node in vector witch custom op via overwriting.
+    auto insert_position = node_indexes[0];
+
+    auto new_node = model().mutable_graph()->mutable_node(insert_position);
+    new_node->Clear();
+    new_node->set_op_type(new_op_name);
+    for (auto input : inputs)
+    {
+        new_node->add_input(input);
+    }
+    for (auto output : outputs)
+    {
+        new_node->add_output(output);
+    }
+
+    // 8. remove remaining nodes from the graph.
+    node_indexes.erase(node_indexes.begin());
+    remove_nodes(node_indexes);
+}
+
+void onnx_import::ONNXModelEditor::ONNXModelEditor::remove_nodes(
+    const std::vector<int>& nodes_to_remove)
+{
+    auto nodes = model().mutable_graph()->mutable_node();
+
+    int idx = -1;
+    const auto discard_node = [&idx, &nodes_to_remove](const ngraph_onnx::NodeProto&) {
+        ++idx;
+        return std::count(nodes_to_remove.begin(), nodes_to_remove.end(), idx);
```

Comment:
Same suggestion as above - consider `any_of` here too. Also you can start from `idx = 0`, use `idx++` when calling `any_of` and this lambda will become a one-liner.